subprojects/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java

```diff
@@ -194,25 +202,15 @@ private List<String> sourceNames() {
 
     private void executeExternalJavadoc(StandardJavadocDocletOptions options) {
         JavadocSpec spec = new JavadocSpec();
-        spec.setExecutable(getExecutable());
+        // Note, we do not set an executable on the spec from the executable field,
+        // because it is always projected from the javadocTool
         spec.setOptions(options);
         spec.setIgnoreFailures(!isFailOnError());
         spec.setWorkingDir(getProjectLayout().getProjectDirectory().getAsFile());
         spec.setOptionsFile(getOptionsFile());
 
-        if (spec.getExecutable() != null) {
-            new JavadocGenerator(getExecActionFactory()).execute(spec);
-        } else {
-            getJavadocToolAdapter().execute(spec);
-        }
-    }
-
-    private JavadocToolAdapter getJavadocToolAdapter() {
-        if (javadocTool.isPresent()) {
-            return (JavadocToolAdapter) this.javadocTool.get();
-        }
-        JavaToolchainSpec explicitToolchain = new CurrentJvmToolchainSpec(getObjectFactory());
-        return (JavadocToolAdapter) getJavaToolchainService().javadocToolFor(explicitToolchain).get();
+        JavadocToolAdapter javadocToolAdapter = (JavadocToolAdapter) getJavadocTool().get();
+        javadocToolAdapter.execute(spec);
```

Comment:
This is an additional breaking change, because Javadoc allowed to specify both the toolchain tool and the executable, and then the executable had a higher priority. After the change, the priority will be unified with other tasks, meaning the executable has lower priority than direct tool property assignment, but still higher priority than the java extension toolchain.